amqp/channelpool/pool.go

```diff
@@ -154,6 +163,14 @@ func (p *pool) Get(ctx context.Context) (*amqp.Channel, error) {
 	}
 
 	pe := e.(*poolEntity)
+
+	pe.mu.RLock()
+	defer pe.mu.RUnlock()
+
+	if pe.err != nil {
+		return nil, err
+	}
```

Comment:
nit: factorize in a `poolEntity.Error()` method

Reply:
I also need to handle closed here.